java/cdk/src/test/java/sleeper/cdk/UtilsVersionTest.java

```diff
@@ -17,17 +17,69 @@
 package sleeper.cdk;
 
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import sleeper.configuration.properties.InstanceProperties;
+import sleeper.configuration.properties.local.SaveLocalProperties;
 
 import java.io.IOException;
+import java.nio.file.Path;
 import java.util.regex.Pattern;
+import java.util.stream.Stream;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatCode;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static sleeper.cdk.Utils.getVersion;
+import static sleeper.cdk.UtilsTestHelper.cdkContextWithPropertiesFile;
+import static sleeper.cdk.UtilsTestHelper.cdkContextWithPropertiesFileAndSkipVersionCheck;
+import static sleeper.cdk.UtilsTestHelper.createInstancePropertiesWithVersion;
 
 public class UtilsVersionTest {
+    @TempDir
+    private Path tempDir;
 
     @Test
-    void shouldGetVersionAsResource() throws IOException {
-        assertThat(Utils.getVersion())
+    void shouldGetVersionAsResource() {
+        assertThat(getVersion())
                 .matches(Pattern.compile("\\d+\\.\\d+\\.\\d+(-SNAPSHOT)?"));
     }
```

Comment:
There's also a case for when we're deploying a new instance, so there is no deployed version and the system-defined properties won't be set in the local properties.